File: src/my_project/easy_problems/from201to250/rank_transform_array.py
from typing import List 

class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        
        
        ordered_list = list()
        len_arr = len(arr)
        temp_dict = dict()
        
        for i in range(len_arr):
            temp_dict[arr[i]]= 0
            
        for i in temp_dict.keys():
    
            ordered_list.append(i)
            
        ordered_list.sort()
        
        solution_dict = dict()
        value_index = 1        
        for i in ordered_list: 
            solution_dict[i] = value_index 
            value_index += 1
            
        for i in range(len_arr):
            arr[i] = solution_dict[arr[i]]
        
        return arr
    
solution = Solution()

print(solution.arrayRankTransform(arr = [40,10,20,30]))

print(solution.arrayRankTransform(arr = [100,100,100]))

print(solution.arrayRankTransform(arr = [37,12,28,9,100,56,80,5,12]))

File: src/my_project_typescript/easy_problems_typescript/from51To100/rankTransformArray.ts
function arrayRankTransform(arr: number[]): number[] {
    let orderedList: Array<number> = []
    let lenArr: number = arr.length
    let tempDict: any = {}

    for (let i = 0; i < lenArr; i++){
        tempDict[arr[i]] = 0
    }

    for (let i in tempDict){
        orderedList.push(parseInt(i))
    }

    orderedList = orderedList.sort((a,b) => a <= b ?-1:1)

    let solutionDict: any = {}
    let valueIndex = 1

    for (let j of orderedList){
        solutionDict[j] = valueIndex
        valueIndex += 1
    }

    for (let j = 0; j < lenArr; j++){
        arr[j] = solutionDict[arr[j]]
    }

    return arr
};


console.log(arrayRankTransform([40,10,20,30]))

console.log(arrayRankTransform([100,100,100]))

console.log(arrayRankTransform([37,12,28,9,100,56,80,5,12]))
